app/assets/javascripts/discourse/app/components/d-lightbox.js

```diff
@@ -0,0 +1,465 @@
+import {
+  ANIMATION_DURATION,
+  KEYBOARD_SHORTCUTS,
+  LAYOUT_TYPES,
+  LIGHTBOX_APP_EVENT_NAMES,
+  LIGHTBOX_ELEMENT_ID,
+  SWIPE_DIRECTIONS,
+  TITLE_ELEMENT_ID,
+} from "discourse/lib/lightbox/constants";
+import {
+  createDownloadLink,
+  getSwipeDirection,
+  openImageInNewTab,
+  preloadItemImages,
+  scrollParentToElementCenter,
+  setCarouselScrollPosition,
+  setSiteThemeColor,
+} from "discourse/lib/lightbox/helpers";
+
+import Component from "@glimmer/component";
+import { bind } from "discourse-common/utils/decorators";
+import discourseLater from "discourse-common/lib/later";
+import { htmlSafe } from "@ember/template";
+import { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+
+export default class DLightbox extends Component {
+  @service appEvents;
+
+  @tracked callbacks = {};
+  @tracked items = [];
+  @tracked options = {};
+
+  @tracked isVisible = false;
+  @tracked isLoading = false;
+  @tracked willClose = false;
+
+  @tracked currentIndex = 0;
+  @tracked currentItem = {};
+
+  @tracked isZoomed = false;
+  @tracked isRotated = false;
+  @tracked isFullScreen = false;
+  @tracked rotationAmount = 0;
+
+  @tracked hasCarousel = false;
+  @tracked hasExpandedTitle = false;
+
+  elementId = LIGHTBOX_ELEMENT_ID;
+  titleElementId = TITLE_ELEMENT_ID;
+  animationDuration = ANIMATION_DURATION;
+
+  get layoutType() {
+    return window.innerWidth > window.innerHeight
+      ? LAYOUT_TYPES.HORIZONTAL
+      : LAYOUT_TYPES.VERTICAL;
+  }
+
+  get CSSVars() {
+    const base = "--d-lightbox-image";
+
+    const variables = [
+      `${base}-animation-duration: ${this.animationDuration}ms;`,
+    ];
+
+    if (!this.currentItem) {
+      return htmlSafe(variables.join(""));
+    }
+
+    const { width, height, aspectRatio, dominantColor, fullsizeURL, smallURL } =
+      this.currentItem;
+
+    variables.push(
+      `${base}-rotation: ${this.rotationAmount}deg`,
+      `${base}-width: ${width}px`,
+      `${base}-height: ${height}px`,
+      `${base}-aspect-ratio: ${aspectRatio}`,
+      `${base}-dominant-color: #${dominantColor}`,
+      `${base}-full-size-url: url(${fullsizeURL})`,
+      `${base}-small-url: url(${smallURL})`
+    );
+
+    return htmlSafe(variables.filter(Boolean).join(";"));
```

Comment:
Do we need to escape the URLs here to make sure it can't be used an an XSS vector?